File: 백준/Silver/11057. 오르막 수/오르막 수.java
import java.io.*;

public class Main {
    static final int MOD = 10007;

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine().trim());

        long result = 1;

        for (int i = 1; i <= 9; i++) {
            result = (result * (n + i)) % MOD;
        }

        long denom = 1;
        for (int i = 1; i <= 9; i++) {
            denom = (denom * i) % MOD;
        }

        result = (result * modPow(denom, MOD - 2)) % MOD;

        System.out.println(result);
    }

    static long modPow(long a, int b) {
        long res = 1;
        while (b > 0) {
            if ((b & 1) == 1) res = (res * a) % MOD;
            a = (a * a) % MOD;
            b >>= 1;
        }
        return res;
    }
}

File: 백준/Silver/11057. 오르막 수/오르막 수.rs
use std::io::{self, Read};

const MOD: i64 = 10007;

fn mod_pow(mut a: i64, mut b: i64) -> i64 {
    let mut res: i64 = 1;
    while b > 0 {
        if b & 1 == 1 {
            res = (res * a) % MOD;
        }
        a = (a * a) % MOD;
        b >>= 1;
    }
    res
}

fn main() {
    let mut input = String::new();
    io::stdin().read_to_string(&mut input).unwrap();
    let n: i64 = input.trim().parse().unwrap();

    let mut result: i64 = 1;
    for i in 1..=9 {
        result = (result * (n + i)) % MOD;
    }

    let mut denom: i64 = 1;
    for i in 1..=9 {
        denom = (denom * i) % MOD;
    }

    result = (result * mod_pow(denom, MOD - 2)) % MOD;

    println!("{}", result);
}